tools/twix/src/panels/plot.rs

```diff
@@ -211,76 +291,37 @@ impl Widget for &mut PlotPanel {
-                            let content = latest_value
-                                .and_then(|value| {
-                                    to_string_pretty(&value).wrap_err("failed to prettify value")
-                                })
-                                .unwrap_or_else(|error| error.to_string());
-                            ui.label(content);
-                            let code_edit = TextEdit::multiline(&mut line_data.lua_text)
-                                .font(TextStyle::Monospace)
-                                .code_editor()
-                                .lock_focus(true);
-                            if ui.add(code_edit).changed() {
-                                line_data.lua_error = match line_data
-                                    .lua
-                                    .load(&line_data.lua_text)
-                                    .eval::<Function>()
-                                {
-                                    Ok(function) => {
-                                        line_data
-                                            .lua
-                                            .globals()
-                                            .set("conversion_function", function)
-                                            .unwrap();
-                                        None
-                                    }
-                                    Err(error) => Some(format!("{error:#}")),
-                                };
-                            }
-                            if let Some(error) = &line_data.lua_error {
-                                ui.colored_label(Color32::RED, error);
-                            } else if let Ok(value) = get_latest_value(&line_data.value_buffer) {
-                                let lua_function: Function =
-                                    line_data.lua.globals().get("conversion_function").unwrap();
-                                let value =
-                                    lua_function.call::<_, f64>(line_data.lua.to_value(&value));
-                                match value {
-                                    Ok(value) => {
-                                        ui.label(value.to_string());
-                                    }
-                                    Err(error) => {
-                                        ui.colored_label(Color32::RED, error.to_string());
-                                    }
-                                }
-                            }
-                        });
-                    });
                 !delete_button.clicked()
             })
             .inner
         });
 
+        if ui.button("✚").clicked() {
+            self.line_datas.push(LineData::new(
+                DEFAULT_LINE_COLORS
+                    .get(self.line_datas.len())
+                    .copied()
+                    .unwrap_or(Color32::TRANSPARENT),
```

Comment:
I think this is YAGNI ! You need at least 11 (?) different plot lines to trigger this case. I'm fine with just having a fallback and showing the user that a random color will be assigned here, instead of silently (and confusingly) reuse already existing colors.

Reply:
Ah, didn't know transparent meant random color.
Resolving